### PR TITLE
Flag for halting on slippage error that defaults to False

### PR DESCRIPTION
**Pull request by slundqui**
Also adding traceback to crash report

### CODE REVIEW

File: lib/agent0/agent0/hyperdrive/exec/trade_loop.py [hunk 1]
```diff
@@ -69,14 +76,28 @@ def trade_if_new_block(
...
                 )
             elif trade_result.status == TradeStatus.FAIL:
-                logging.info(
-                    "AGENT %s (%s) attempted %s for %g\nCrashed with error: %s",
-                    str(trade_result.agent.checksum_address),
-                    trade_result.agent.policy.__class__.__name__,
-                    trade_result.trade_object.market_action.action_type,
-                    float(trade_result.trade_object.market_action.trade_amount),
-                    trade_result.exception,
+                is_slippage = isinstance(trade_result.exception, ContractCustomError) and (
```
Comment (dpaiton) [inline lib/agent0/agent0/hyperdrive/exec/trade_loop.py:79]: did you confirm with Alex that this is always bc of slippage?

File: lib/agent0/agent0/hyperdrive/exec/trade_loop.py [hunk 2]
```diff
@@ -69,14 +76,28 @@ def trade_if_new_block(
...
+                    "OutputLimit raised" in trade_result.exception.args[1]
                 )
+                if is_slippage:
+                    logging.info(
```
Comment (dpaiton) [inline lib/agent0/agent0/hyperdrive/exec/trade_loop.py:83]: I think this should be a warning, right?

File: lib/agent0/agent0/hyperdrive/exec/trade_loop.py [hunk 3]
```diff
@@ -69,14 +76,28 @@ def trade_if_new_block(
...
+                        "AGENT %s (%s) attempted %s for %g\nSlippage detected: %s",
+                        str(trade_result.agent.checksum_address),
+                        trade_result.agent.policy.__class__.__name__,
+                        trade_result.trade_object.market_action.action_type,
+                        float(trade_result.trade_object.market_action.trade_amount),
+                        trade_result.exception,
+                    )
+                else:
+                    logging.info(
```
Comment (dpaiton) [inline lib/agent0/agent0/hyperdrive/exec/trade_loop.py:92]: logging.error?

File: lib/agent0/agent0/hyperdrive/exec/trade_loop.py (hunk 1 above)
Comment (slundqui) [reply lib/agent0/agent0/hyperdrive/exec/trade_loop.py:79]: Confirmed

**Review by dpaiton — APPROVED** [review]

**Pull request merged**